lib/src/solana_wallet.dart

```diff
@@ -1,35 +1,58 @@
 import 'package:bip39/bip39.dart' as bip39;
 import 'package:cryptography/cryptography.dart';
 import 'package:solana/src/base58/base58.dart' as base58;
+import 'package:solana/src/types/hd_key_pair.dart';
 
 /// Signs transactions to be sent to a Solana Cluster
 class SolanaWallet {
-  SolanaWallet._fromKeyPair(KeyPair keyPair, this.address) : _keyPair = keyPair;
-
-  static final _ed25519 = Ed25519();
+  SolanaWallet._fromKeyPair(
+    this._keyPair, {
+    required this.address,
+  });
 
   /// Creates and initializes a new SolanaWallet for the given bip39
   /// [mnemonic] string of 12 words
-  static Future<SolanaWallet> fromMnemonic(String mnemonic) async {
+  static Future<SolanaWallet> fromMnemonic(
+    String mnemonic, {
+    int? account,
+  }) async {
     // Create the seed
     final List<int> seedBytes = bip39.mnemonicToSeed(mnemonic);
-    // Create a private key from the seed
-    final SecretKeyData privateKey = SecretKeyData(seedBytes.sublist(0, 32));
-    // Build the key pair
-    final keyPair = await _ed25519.newKeyPairFromSeed(privateKey.bytes);
-    final address = base58.encode((await keyPair.extractPublicKey()).bytes);
-    // Finally, create a new wallet
-    return SolanaWallet._fromKeyPair(keyPair, address);
+    final KeyPair keyPair = await _deriveKeyPair(seedBytes, account);
+    final PublicKey publicKey = await keyPair.extractPublicKey();
+    if (publicKey is SimplePublicKey) {
+      // Finally, create a new wallet
+      return SolanaWallet._fromKeyPair(
+        keyPair,
+        address: base58.encode(publicKey.bytes),
+      );
+    } else {
+      throw Exception('could not build a key pair');
+    }
   }
 
   /// Returns a Future that resolves to the result of signing
   /// [data] with the private key held internally by a given
   /// instance
-  Future<Signature> sign(List<int> data) =>
+  Future<Signature> sign(List<int> data) async =>
       _ed25519.sign(data, keyPair: _keyPair);
 
   /// The address or public key of this wallet
+  static final _ed25519 = Ed25519();
+  final KeyPair _keyPair;
   final String address;
 
-  final KeyPair _keyPair;
+  static String _getHDPath(int account) => "m/44'/501'/${account - 1}'";
```

Comment:
I would rather place it into `HDKeyPair` class and let it accept `account`. It seems to be an implementation detail.

Reply:
It is. I just didn't move it. Thanks for pointing it out!

Reply:
My previous comment was wrong. In fact. This is intended. The HD path is not part of the derivation or the key pair. It's in fact a parameter to the HD key pair object. Users of the library could want to use it for different tokens (in that case they would change the 501 in the path.

So basically. The derivation path is part of the wallet and not the key pair.

Reply:
Yeah, makes sense.